File: Контест 3/F.py
def good(cnt, q, s, t):
  return t // q + t // s >= cnt

n, quick, slow = map(int, input().split())
if quick > slow:
  quick, slow = slow, quick

l = 0
r = (n-1) * slow
while r - l > 1:
  m = (l + r) // 2
  if not good(n - 1, quick, slow, m):
    l = m
  else:
    r = m
print(r + quick)

File: Контест 3/f.cpp
#include <iostream>
#include <vector>

bool good(int cnt, int q, int s, int t)
{
  return t / q + t / s >= cnt;
}

int main()
{
  int n, quick, slow;
  std::cin >> n >> quick >> slow;
  if (quick > slow){
    std::swap(quick, slow);
  }
  int l = 0; 
  int r = (n - 1) * slow;

  while (r - l > 1)
  {
    int m = (l + r) / 2;
    if (!(good(n - 1, quick, slow, m)))
    {
      l = m;
    }
    else
    {
      r = m;
    }
  }
  std::cout << r + quick << std::endl;
  return 0;
}